File: motor3d 2.0/AristaGrafo.vb
Namespace Discreta
    Public Class Arista
        Private mPrimerVertice As Long
        Private mSegundoVertice As Long

        Public ReadOnly Property PrimerVertice As Long
            Get
                Return mPrimerVertice
            End Get
        End Property

        Public ReadOnly Property SegundoVertice As Long
            Get
                Return mSegundoVertice
            End Get
        End Property

        Public Sub New(ByVal PrimerVertice As Long, ByVal SegundoVertice As Long)
            If PrimerVertice >= 0 Then
                If SegundoVertice > 0 Then
                    mPrimerVertice = PrimerVertice
                    mSegundoVertice = SegundoVertice
                Else
                    Throw New ExcepcionDiscreta("ARISTA (NEW): El segundo vertice no puede ser menor que cero" & vbNewLine & _
                                                "Primer vertice=" & PrimerVertice.ToString & vbNewLine & _
                                                "Segundo vertice=" & SegundoVertice.ToString)
                End If
            Else
                Throw New ExcepcionDiscreta("ARISTA (NEW): El primer vertice no puede ser menor que cero" & vbNewLine & _
                                                "Primer vertice=" & PrimerVertice.ToString & vbNewLine & _
                                                "Segundo vertice=" & SegundoVertice.ToString)
            End If
        End Sub
    End Class
End Namespace

File: motor3d 2.0.ConvertedToC#/AristaGrafo.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Xml.Linq;
namespace Motor3D.Discreta
{
	public class Arista
	{
		private long mPrimerVertice;

		private long mSegundoVertice;
		public long PrimerVertice {
			get { return mPrimerVertice; }
		}

		public long SegundoVertice {
			get { return mSegundoVertice; }
		}

		public Arista(long PrimerVertice, long SegundoVertice)
		{
			if (PrimerVertice >= 0) {
				if (SegundoVertice > 0) {
					mPrimerVertice = PrimerVertice;
					mSegundoVertice = SegundoVertice;
				} else {
					throw new ExcepcionDiscreta("ARISTA (NEW): El segundo vertice no puede ser menor que cero" + Constants.vbNewLine + "Primer vertice=" + PrimerVertice.ToString() + Constants.vbNewLine + "Segundo vertice=" + SegundoVertice.ToString());
				}
			} else {
				throw new ExcepcionDiscreta("ARISTA (NEW): El primer vertice no puede ser menor que cero" + Constants.vbNewLine + "Primer vertice=" + PrimerVertice.ToString() + Constants.vbNewLine + "Segundo vertice=" + SegundoVertice.ToString());
			}
		}
	}
}

